src/pages/Main/index.tsx

```diff
@@ -0,0 +1,51 @@
+import {
+  StyleSheet,
+  Text,
+  View,
+  Image,
+  ScrollView,
+  TouchableHighlight,
+} from 'react-native';
+import React, {useState} from 'react';
+import CryptCards from '../../components/CryptCards';
+import {
+  ContainerView,
+  HeaderImage,
+  HeaderText,
+  HeaderView,
+  ListScrollView,
+  TouchableText,
+  WarnText,
+} from './styles';
+
+const Main = () => {
+  const [open, setOpen] = useState(false);
+
+  const toggleCards = () => {
+    setOpen(!open);
```

Comment:
this can fail, because useState is async, 

you shoud use the previous value in this way:

setOpen((prev) => !prev)